resolver-dns/src/main/java/io/netty/resolver/dns/DnsServerAddresses.java

```diff
@@ -106,15 +104,15 @@ public static DnsServerAddresses shuffled(InetSocketAddress... addresses) {
         return shuffled0(sanitize(addresses));
     }
 
-    private static DnsServerAddresses shuffled0(final InetSocketAddress[] addresses) {
-        if (addresses.length == 1) {
-            return singleton(addresses[0]);
+    private static DnsServerAddresses shuffled0(List<InetSocketAddress> addresses) {
+        if (addresses.size() == 1) {
+            return singleton(addresses.get(0));
         }
 
         return new DefaultDnsServerAddresses("shuffled", addresses) {
             @Override
             public DnsServerAddressStream stream() {
-                return new ShuffledDnsServerAddressStream(addresses);
+                return new ShuffledDnsServerAddressStream(addresses.toArray(new InetSocketAddress[0]));
```

Comment:
In the interest of avoiding conversions, if we are going to change from array to `List` can we either:
- make this change consistent across stream types (e.g. also to `ShuffledDnsServerAddressStream`)
- do the conversion once (instead of on each call to `stream()`)